### PR TITLE
Hdf2ckl profile w dependent fix

### LINKED ISSUE #1472: saf convert hdf2ckl does not produce a valid CKL file for InSpec results from profiles with dependent profiles.
SAF CLI 1.2.15

Running an InSpec profile with dependent profiles to produce a JSON result to then convert to CKL incorrectly incorporates the duplicated "parent" profile controls into the CKL.

Take the profile here for example:  https://github.com/vmware/dod-compliance-and-automation/tree/master/vsphere/8.0/vsphere/inspec/vmware-vsphere-8.0-stig-baseline

The JSON InSpec outputs for whatever reason puts a copy of all of the controls from the dependent profiles in the parent profile with no results. Heimdall correctly filters this extra data out when loading these results.

The first profile in the JSON seems to always be the parent as well.

Running this
`saf convert hdf2ckl -i C:\inspec\Reports\Runner\MyvSphere8Report.json -o C:\inspec\Reports\Runner\MyvSphere8Report.ckl --hostname 10.186.30.81 --fqdn myvcenter.local --ip 10.186.30.81 --mac 00:00:00:00:00:00`

After import into STIG Viewer you can see there are 336 controls where there should be 168 so its adding 168 with a status of not reviewed.
image

Exporting from Heimdall it looks like this which is correct
image

I attached the json results and the example CKL files from Heimdall and the manual conversion.

[MyvSphere8Report.zip](https://github.com/mitre/saf/files/11690443/MyvSphere8Report.zip)

### PR DESCRIPTION
**Pull request by vanessuniq**
- Fix hdf2ckl to produce a valid CKL file for InSpec results from profiles with dependent profiles by filtering out the duplicated data.
- Updated the test for hdf2ckl

Fix #1472

### CODE REVIEW

**Comment by aaronlippold** [comment]
Does this account for both types of dependent profiles: 

Wrapper: Wide layer with multiple benchmarks at the same level
Expected: Create multiple checklists or the user can ask for a 'grouped single checklist'
  - https://github.com/vmware/dod-compliance-and-automation/tree/master/vsphere/7.0/vsphere/inspec/vmware-vsphere-7.0-stig-baseline 
  - https://github.com/chris-rock/acme-inspec-profile 
Overlay: Parent/Child/grand-child chain
Expected: Create a single checklist file
  - https://github.com/mitre/aws-rds-crunchy-data-postgresql-9-stig-baseline

File: test/commands/convert/hdf2ckl.test.ts
Comment (em-c-rod) [inline test/commands/convert/hdf2ckl.test.ts:12]: Should we add this as a test rather than remove the previous? Or remove the other file if it is no longer needed?

Comment (vanessuniq) [reply test/commands/convert/hdf2ckl.test.ts:12]: The red_hat file is still being used in the test with metadata. But yeah, we should probably add it as a test for profile with dependents.

**Comment by Hookwitz** [comment]
@aaronlippold this will address the overlay. I do not believe either export functionality (Heimdall or SAF CLI) contains the capability to split into multiple checklists nor provides user input.

**Review by Amndeep7 — APPROVED** [review]
The is good code-wise.  Just waiting for @rlakey to validate it works for him before merge.

**Review by rlakey — APPROVED** [review]

**Pull request merged**
